### PR TITLE
Fix: Consistent UI

### PR DESCRIPTION
**Pull request by jasmussen**
This PR does a number of things in the name of addressing or almost addressing the remaining items in #18667. In additionally, it fixes all items from #19904. Here's what it does:

It makes the buttongroup consistent with regards to toggles:

Screenshot 2020-06-16 at 13 18 45

Screenshot 2020-06-16 at 13 19 51

_Note to third parties, the inactive buttons aren't _secondary_, they're just buttons._

It makes the sidebar tabs consistent with the inserter:

Screenshot 2020-06-16 at 13 19 55

You'll also note in that one, that the text color is now uniformly the same dark gray everywhere in the UI.

It makes all panels the same height, previously some panels were 50px tall, others 48, some arbitrary multiples. It also removes a $panel-padding variable, so as to encourage the grid system instead:
 
Screenshot 2020-06-16 at 11 43 54

It adds a new small close icon, and uses it for both sidebar and prepublish panels. It also vertically aligns it to the chevrons:

Screenshot 2020-06-16 at 13 24 38

Screenshot 2020-06-16 at 13 24 44

It removes the I/O icons from the form toggle, because they added complexity that wasn't helping convey the active/inactive state:

Screenshot 2020-06-16 at 13 25 37

It fixes so this chevron isn't scaled down:

Screenshot 2020-06-16 at 13 26 30

I would like to do much more to improve things, including do an audit of button style usage in the sidebar — small, secondary, tertiary, labels, etc. — for all blocks, including doing some positioning improvements. But instead of creating one mega PR, I'm breaking it up.

Would appreciate your thoughts.

### CODE REVIEW

File: packages/edit-post/src/components/sidebar/settings-header/style.scss [hunk 1]
```diff
@@ -1,29 +1,15 @@
...
 	border: none;
 	box-shadow: none;
 	cursor: pointer;
 	display: inline-block;
 	padding: 3px 15px; // Use padding to offset the is-active border, this benefits Windows High Contrast mode
 	margin-left: 0;
-	font-weight: 400;
-	color: $dark-gray-900;
+	font-weight: 500;
+	color: $dark-gray-primary;
```
Comment (youknowriad) [inline packages/edit-post/src/components/sidebar/settings-header/style.scss:12]: Can't we remove the color and rely on the inheritance instead or is this really needed?

File: packages/edit-post/src/components/sidebar/settings-header/style.scss [hunk 2]
```diff
@@ -41,8 +27,8 @@
 	&.is-active {
 		// The transparent shadow ensures no jumpiness when focus animates on an active tab.
 		box-shadow: inset 0 0 0 $border-width-focus transparent, inset 0 0 -$border-width-tab 0 0 var(--wp-admin-theme-color);
-		font-weight: 600;
 		position: relative;
+		z-index: z-index(".edit-post-sidebar__panel-tab.is-active");
```
Comment (youknowriad) [inline packages/edit-post/src/components/sidebar/settings-header/style.scss:31]: Why do we need a z-index?

File: packages/editor/src/components/post-publish-panel/index.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ import {
 } from '@wordpress/components';
 import { withSelect, withDispatch } from '@wordpress/data';
 import { compose } from '@wordpress/compose';
-import { close } from '@wordpress/icons';
+import { closeSmall } from '@wordpress/icons';
```
Comment (youknowriad) [inline packages/editor/src/components/post-publish-panel/index.js:20]: Should we replace all close icons or do you think there's a need for a small and a big one?

**Review by youknowriad — COMMENTED** [review]
While I didn't test deeply, by looking at the code, the changes seem safe for me.

**Comment by youknowriad** [comment]
Also, let's try to not add new changes and use separate PRs to make things easy to test. Ideally, each change happen in its own PR.

File: packages/edit-post/src/components/sidebar/settings-header/style.scss (hunk 2 above)
Comment (jasmussen) [reply packages/edit-post/src/components/sidebar/settings-header/style.scss:31]: Screenshot 2020-06-16 at 15 09 34

It makes the tab border more visible

File: packages/editor/src/components/post-publish-panel/index.js (hunk 3 above)
Comment (jasmussen) [reply packages/editor/src/components/post-publish-panel/index.js:20]: I think there's a use case for both, honestly, it's okay they're big in the modal dialogs, but in the sidebar contexts they should not be quite as notable.

**Comment by youknowriad** [comment]
in my testing, the toggle with I/O feels a bit too wide (especially when it's unchecked), do you think we can reduce its width?

**Comment by youknowriad** [comment]
Its border width also feels inconsistent with the inputs... a 1px border might look better IMO.

**Comment by youknowriad** [comment]
>  1px border might look better IMO.

This is also the same for checkbox so we might want to do it separately if ever.

**Comment by jasmussen** [comment]
I updated the margin to match that of the checkbox.

> Its border width also feels inconsistent with the inputs... a 1px border might look better IMO.

I think I agree, but I'd like to polish both the toggle and checkbox and radio separately, they're all 2px now.

**Comment by youknowriad** [comment]
> I updated the margin to match that of the checkbox.

I actually was thinking about the toggle itself :) 

Regardless this PR looks good.

**Review by youknowriad — APPROVED** [review]

**Comment by jasmussen** [comment]
> I actually was thinking about the toggle width itself :)

Ah. Yep, I've seen a G2 design that I want to look at. Thanks.

**Comment by gziolo** [comment]
I see failing e2e tests that block this PR. It’s a recurring issue that visual changes break them. We need to rethink approach how user interactions are coded, my guess is that tests fail when mouse cursor is moved. @youknowriad, could we limit mouse interactions in e2e tests to absolute minimum and use keyboard interactions instead?

**Comment by jasmussen** [comment]
Thanks so much for looking, Gzregorz. I'm banging my head against the tests on this one, I'd love to learn how to fix them, I've definitely fixed a few in the past. But these confuse me to no end. It's the same, frankly, with https://github.com/WordPress/gutenberg/pull/22213, which is a great code quality improvement that would be nice to land also.

**Comment by jasmussen** [comment]
Tests are passing! Turns out there are two ways to update snapshots and I was only familiar with the one: `npm run test-unit -- --updateSnapshot`. This one needed running: `npm run test-e2e -- -u`

**Pull request merged**